server/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java

```diff
@@ -68,23 +68,31 @@ public class EsThreadPoolExecutor extends ThreadPoolExecutor {
     @Override
     public void execute(Runnable command) {
         final Runnable wrappedRunnable = wrapRunnable(command);
-        try {
-            super.execute(wrappedRunnable);
-        } catch (Exception e) {
-            if (wrappedRunnable instanceof AbstractRunnable abstractRunnable) {
-                try {
-                    // If we are an abstract runnable we can handle the exception
-                    // directly and don't need to rethrow it, but we log and assert
-                    // any unexpected exception first.
-                    if (e instanceof EsRejectedExecutionException == false) {
-                        logException(abstractRunnable, e);
+        while (true) {
+            try {
+                super.execute(wrappedRunnable);
+                return;
+            } catch (EsExecutors.RetryRejectedExecutionException e) {
+                // only used with ExecutorScalingQueue and ForceQueuePolicy.
+                // yield to increase hope of retry working.
+                Thread.yield();
```

Comment:
Are there any problems caused by possible thread starvation here, if one or more threads persistently fails to queue an item?